fedbadges/utils.py

```diff
@@ -111,13 +112,25 @@ def notification_callback(topic, msg):
 def user_exists_in_fas(config, user):
     """ Return true if the user exists in FAS. """
     default_url = 'https://admin.fedoraproject.org/accounts/'
+    if config.get["freeipa-fas"]:
+        return user_exists_in_freeipa_fas(config, user)
+
     fas2 = fedora.client.AccountSystem(
         base_url=config['fas_credentials'].get('base_url', default_url),
         username=config['fas_credentials']['username'],
         password=config['fas_credentials']['password'],
     )
     return bool(fas2.person_by_username(user))
 
+def user_exists_in_freeipa_fas(config, user):
+    """ Return true if the user exists in freeipa-fas. """
+    default_url = 'https://admin.fedoraproject.org/accounts/'
+    client = Client(
+        url=config['fas_credentials'].get('base_url', default_url)
+    )
+    user = client.get_user(username=user).result
```

Comment:
With the new API, you don't need both `.result` and `has_key("result")`. The `.result` property gives you directly acces to the result dictionnary or list, and if it's not found it will raise a `fasjson_client.errors.APIError`.